module/zfs/spa.c

```diff
@@ -3326,6 +3338,103 @@ spa_start_livelist_condensing_thread(spa_t *spa)
 	    spa_livelist_condense_cb, spa, minclsyspri);
 }
 
+static void
+spa_sync_time_logger(spa_t *spa, dmu_tx_t *tx)
+{
+	uint64_t curtime;
+
+	if (!spa_feature_is_enabled(spa, SPA_FEATURE_TXG_TIMELOG)) {
+		return;
+	}
+	if (!spa_writeable(spa)) {
+		return;
+	}
+	if (tx->tx_txg == spa->spa_last_noted_txg) {
+		return;
+	}
+	curtime = gethrestime_sec();
+	if (curtime < spa->spa_last_noted_txg_time + spa_note_txg_time) {
+		return;
+	}
+
+	spa->spa_last_noted_txg_time = curtime;
+	spa->spa_last_noted_txg = tx->tx_txg;
+
+	mutex_enter(&spa->spa_txg_log_time_lock);
+	dbrrd_add(&spa->spa_txg_log_time, curtime, tx->tx_txg);
+
+	if (curtime < spa->spa_last_flush_txg_time + spa_flush_txg_time) {
+		goto out;
+	}
+	spa->spa_last_flush_txg_time = curtime;
+
+	if (zap_contains(spa_meta_objset(spa), DMU_POOL_DIRECTORY_OBJECT,
+	    DMU_POOL_TXG_LOG_TIME_MINUTES) == ENOENT) {
+		VERIFY0(zap_add(spa_meta_objset(spa),
+		    DMU_POOL_DIRECTORY_OBJECT, DMU_POOL_TXG_LOG_TIME_MINUTES,
+		    1, sizeof (spa->spa_txg_log_time.dbr_minutes),
+		    &spa->spa_txg_log_time.dbr_minutes, tx));
+		VERIFY0(zap_add(spa_meta_objset(spa),
+		    DMU_POOL_DIRECTORY_OBJECT, DMU_POOL_TXG_LOG_TIME_DAYS,
+		    1, sizeof (spa->spa_txg_log_time.dbr_days),
+		    &spa->spa_txg_log_time.dbr_days, tx));
+		VERIFY0(zap_add(spa_meta_objset(spa),
+		    DMU_POOL_DIRECTORY_OBJECT, DMU_POOL_TXG_LOG_TIME_MONTHS,
+		    1, sizeof (spa->spa_txg_log_time.dbr_months),
+		    &spa->spa_txg_log_time.dbr_months, tx));
```

Comment:
You are storing the data as array of bytes. Where is byteswap handling?  We already have BRT non-endian safe that must be fixed, please don't add more.